common-util/List/ListTable/AgentHashLink.tsx

```diff
@@ -0,0 +1,46 @@
+import { Skeleton } from 'antd';
+import { useEffect, useState } from 'react';
+
+import { AddressLink, NA } from '@autonolas/frontend-library';
+
+import { getMetadataHashByServiceId } from 'common-util/functions/requests';
+
+const FETCH_HASH_TIMEOUT = 10000;
+
+type AddressHashLinkProps = {
+  serviceId: string;
+  suffixCount: number;
+  canCopy: boolean;
+  textMinWidth: number;
+  chainId: number;
+};
+
+export const AddressHashLink = ({ serviceId, ...rest }: AddressHashLinkProps) => {
+  const [text, setText] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const getMetaDataHash = async () => {
+      try {
+        // Create an AbortController for each request
+        const controller = new AbortController();
+        const signal = controller.signal;
+
+        // Set a timeout to abort the request if it takes too long
+        const timeoutId = setTimeout(() => controller.abort(), FETCH_HASH_TIMEOUT);
```

Comment:
sometimes for some hashes it takes too long to get the response, so adding cancellation not to show loader forever